packages/insomnia-app/app/ui/components/error-boundary.js

```diff
@@ -24,6 +27,17 @@ class SingleErrorBoundary extends React.PureComponent<Props, State> {
     };
   }
 
+  // eslint-disable-next-line camelcase
+  UNSAFE_componentWillReceiveProps(nextProps: $ReadOnly<Props>) {
+    const { error, info } = this.state;
+
+    if (nextProps.invalidateWith === undefined || (error === null && info === null)) {
+      return;
+    }
+
+    this.setState({ error: null, info: null });
```

Comment:
I feel like we should be checking if the invalidation key changed here. I know your current solution works, but it's hard to follow what's happening without adding a ton of comments. I suggest something a bit more descriptive and explicit.

```suggestion
    const invalidationKeyChanged = nextProps.invalidateWith !== this.props.invalidateWith;
    const isErrored = error !== null || info !== null;
    const shouldResetError = invalidationKeyChanged && isErrored;
    
    if (shouldResetError) {
        this.setState({ error: null, info: null });
    }
```